syft/workers/base.py

```diff
@@ -425,19 +431,19 @@ def handle_object_msg(self, obj_msg: ObjectMessage):
             obj: a Framework Tensor or a subclass of an AbstractTensor with an id
         """
         obj = obj_msg.object
+        placeholder_id = obj_msg.placeholder_id
 
-        self.object_store.set_obj(obj)
-
-        if isinstance(obj, FrameworkTensor):
-            tensor = obj
-            if (
-                tensor.requires_grad
-                and tensor.origin is not None
-                and tensor.id_at_origin is not None
-            ):
-                tensor.register_hook(
-                    tensor.trigger_origin_backward_hook(tensor.origin, tensor.id_at_origin)
-                )
+        if isinstance(obj, FrameworkTensor) and (
+            obj.requires_grad and obj.origin is not None and obj.id_at_origin is not None
+        ):
+            obj.register_hook(obj.trigger_origin_backward_hook(obj.origin, obj.id_at_origin))
+
+        # Check if we received an object from a protocol run
+        if placeholder_id in self._protocol_placeholders:
+            self._protocol_placeholders[placeholder_id].instantiate(obj)
+        else:
+            # TODO is it really in the else here or do we want to register anyway?
```

Comment:
Seems like it depends on whether the `Placeholders` are already registered in the `ObjectStore`? I'm not sure if they are. If so, then it doesn't seem like we'd need to register them again. If not, maybe this is covered by the `load/store` methods from @Prtfw's PR?

Reply:
I'm with you on interacting consistently, but not on "_the_ object store." I'm still thinking of `Executions` having their own `ObjectStore`, which I imagine being where `Placeholders` are stored. (Ideally, the `Worker's` `ObjectStore` would never contain `Placeholders`.)

Reply:
I think I see what you mean. I'll try to do that in the next PR 🙂